### PR TITLE
ErrorCode tests

### PR DESCRIPTION
**Pull request by omnivion**
### Information
- JIRA story QPPCT-650.

### Checklist 
- [X] All JUnit tests pass (`mvn clean verify`).
- [X] New unit tests written to cover new functionality.
- [X] Added and updated JavaDocs for non-test classes and methods.
- [X] No local design debt. Do you feel that something is "ugly" after your changes?
- [X] Updated documentation (`README.md`, etc.) depending if the changes require it.

### CODE REVIEW

**Comment by clydet** [comment]
@omnivion What do think about naming test methods as a practice going forward? If not for all tests maybe parameterized tests at a minimum.

```
@DisplayName("Pass enum values to our test method")
class EnumSourceExampleTest {
 
    @DisplayName("Should pass non-null enum values as method parameters")
    @ParameterizedTest(name = "{index} => pet=''{0}''")
    @EnumSource(Pet.class)
```

File: commons/src/test/java/gov/cms/qpp/conversion/ErrorCodeTest.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+		} catch (IllegalStateException exception) {
+			Truth.assertThat(exception).hasMessageThat().isEqualTo(errorCode + " does not support formatting");
+		}
+	}
+
+	@Test
+	void testGetCodeIsUnique() {
+		long count = Arrays.stream(ErrorCode.values()).mapToInt(ErrorCode::getCode).count();
+		long expected = ErrorCode.values().length;
+		Truth.assertThat(count).isEqualTo(expected);
```
Comment (halprin) [inline commons/src/test/java/gov/cms/qpp/conversion/ErrorCodeTest.java]: This just counts the number of enums in `ErrorCode` with no reduction for uniques.  This test will always pass.

File: commons/src/test/java/gov/cms/qpp/conversion/ErrorCodeTest.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+import java.util.Arrays;
+import java.util.UUID;
+
+import com.google.common.truth.Truth;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+
+class ErrorCodeTest {
```
Comment (halprin) [inline commons/src/test/java/gov/cms/qpp/conversion/ErrorCodeTest.java]: We already have an existing `ErrorCodeTest`.  `gov.cms.qpp.conversion.model.error.ErrorCodeTest`.  Looks like it wasn't moved over when `ErrorCode` was moved to `common` or something.  We'll need to integrate it.

**Comment by sfradkin** [comment]
@clydet : I like your suggestion about the test naming.  It may seem verbose, but naming tests like this is the norm in other languages and it makes it easier to both understand the intent of the test and understand test output. @omnivion

**Comment by omnivion** [comment]
@sfradkin @clydet I'm not a huge fan. Documenting tests like that makes them harder to maintain, in my experience. It should be pretty obvious what a test does from the contents.

File: commons/src/main/java/gov/cms/qpp/conversion/model/error/FormattedErrorCode.java [hunk 3]
```diff
@@ -34,13 +34,17 @@ public String toString() {
 
 	@Override
 	public boolean equals(Object o) {
```
Comment (FlexionSonarQube) [inline commons/src/main/java/gov/cms/qpp/conversion/model/error/FormattedErrorCode.java]: MAJOR Reduce the number of returns of this method 4, down to the maximum allowed 3. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1142)

**Comment by FlexionSonarQube** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: commons/src/test/java/gov/cms/qpp/conversion/ErrorCodeTest.java [hunk 4]
```diff
@@ -1,19 +1,58 @@
...
+		} catch (IllegalStateException exception) {
+			Truth.assertThat(exception).hasMessageThat().isEqualTo(errorCode + " does not support formatting");
+		}
+	}
 
 	@Test
+	void testGetCodeIsUnique() {
+		long count = Arrays.stream(ErrorCode.values()).mapToInt(ErrorCode::getCode).count();
+		long expected = ErrorCode.values().length;
+		Truth.assertThat(count).isEqualTo(expected);
```
Comment (halprin) [inline commons/src/test/java/gov/cms/qpp/conversion/ErrorCodeTest.java]: This test just counts the number of enums in ErrorCode with no reduction for uniques. This test will always pass.

File: commons/src/test/java/gov/cms/qpp/conversion/ErrorCodeTest.java [hunk 5]
```diff
@@ -1,19 +1,58 @@
...
+	}	@Test
```
Comment (halprin) [inline commons/src/test/java/gov/cms/qpp/conversion/ErrorCodeTest.java]: This `@Test` is on the same line as the ending of the previous test method.

**Pull request merged**